scipy/stats/tests/test_distributions.py

```diff
@@ -3243,12 +3243,7 @@ def test_signature_inspection_2args(self):
 
     def test_signature_inspection_2args_incorrect_shapes(self):
         # both _pdf and _cdf defined, but shapes are inconsistent: raises
-        try:
-            _distr3_gen(name='dummy')
-        except TypeError:
-            pass
-        else:
-            raise AssertionError('TypeError not raised.')
+        assert_raises(TypeError, _distr3_gen, **dict(name='dummy'))
```

Comment:
This should be the same as (more simply) passing `name='dummy'` directly rather than wrapping in a `**dict(name='dummy')`.